near-sdk/src/store/lazy/mod.rs

```diff
@@ -144,6 +144,137 @@ where
     }
 }
 
+/// An persistent lazily loaded option, that stores a value in the storage.
+///
+/// This will only write to the underlying store if the value has changed, and will only read the
+/// existing value from storage once.
+///
+/// # Examples
+/// ```
+/// use near_sdk::store::LazyOption;
+///
+///# near_sdk::test_utils::test_env::setup();
+/// let mut a = LazyOption::new(b"a", None);
+/// assert_eq!(*a, None);
+///
+/// *a = Some("new value".to_owned());
+/// assert_eq!(a.get(), &Some("new value".to_owned()));
+/// ```
+#[derive(BorshSerialize, BorshDeserialize)]
+pub struct LazyOption<T>
+where
+    T: BorshSerialize,
+{
+    /// Key bytes to index the contract's storage.
+    storage_key: Vec<u8>,
+
+    /// Cached value which is lazily loaded and deserialized from storage.
+    #[borsh_skip]
+    cache: OnceCell<CacheEntry<T>>,
+}
+
+impl<T> LazyOption<T>
+where
+    T: BorshSerialize,
+{
+    /// Returns `true` if the value is present in the storage.
+    pub fn is_some(&self) -> bool {
+        self.cache.get().map_or(false, |cache| cache.value().is_some())
```

Comment:
I agree we shouldn't be duplicating APIs since that would add more maintenance if anything. I think I had these before I implemented `Deref` so forgot that it came for free.